__tests__/commands/install/integration.js

```diff
@@ -55,6 +55,15 @@ test.concurrent('install should not hoist packages above their peer dependencies
   });
 });
 
+test.concurrent('install should resolve peer dependencies from same subtrees', async () => {
+  await runInstall({}, 'peer-dep-same-subtree', async (config): Promise<void> => {
+    expect(JSON.parse(await fs.readFile(`${config.cwd}/node_modules/d/node_modules/a/package.json`)).version).toEqual(
+      '1.1.0',
```

Comment:
shouldn't this equal `1.0.0` since d points to `file:../a-1.0.0` as a dependency? and the top level version of a equal `1.1.0`?

Reply:
Yup, that's why the test was failing :)